src/Spaces/pointspace.jl

```diff
@@ -16,18 +16,21 @@ end
 ClimaComms.device(space::PointSpace) = ClimaComms.device(space.context)
 ClimaComms.context(space::PointSpace) = space.context
 
-@deprecate PointSpace(x::Geometry.LocalGeometry) PointSpace(
-    ClimaComms.SingletonCommsContext(ClimaComms.CPUDevice()),
-    x,
-) false
+PointSpace(x::Geometry.LocalGeometry) = PointSpace(ClimaComms.CPUDevice(), x)
+PointSpace(x::Geometry.AbstractPoint) = PointSpace(ClimaComms.CPUDevice(), x)
+
+function PointSpace(device::ClimaComms.AbstractDevice, x)
+    context = ClimaComms.SingletonCommsContext(device)
+    return PointSpace(context, x)
+end
 
 function PointSpace(
     context::ClimaComms.AbstractCommsContext,
     local_geometry::LG,
 ) where {LG <: Geometry.LocalGeometry}
     FT = Geometry.undertype(LG)
-    # TODO: inherit array type
```

Comment:
@simonbyrne inheriting the array type ends up triggering getindex errors, should we just punt on this for now?

Reply:
I was hoping this would be an easy fix :/

Reply:
Fixed via adapt for now. I think this solution may be fine